indico/web/client/js/react/components/WTFDateTimeField.jsx

```diff
@@ -0,0 +1,237 @@
+      trigger(dateId, 'change');
+      timePickerRef.current.picker.required = value;
+    },
+    [dateField, timeField, dateId, defaultTime, updateTime]
+  );
+
+  useEffect(() => {
+    if (!timezoneField) {
+      return;
+    }
+    function handleTimezone() {
+      setTimezone(timezoneField.value);
+    }
+    timezoneField.addEventListener('change', handleTimezone);
+
+    return () => {
+      timezoneField.removeEventListener('change', handleTimezone);
+    };
+  }, [timezoneField]);
+
+  useEffect(() => {
+    timePickerRef.current.picker.required = required;
+  }, [required]);
+
+  useEffect(() => {
+    if (!linkedField) {
+      return;
+    }
+    function handleDateChange() {
+      const linkedDate = moment(linkedFieldDateElem.value, 'DD/MM/YYYY');
+      if (
+        (linkedField.notBefore && linkedDate.isAfter(date, 'day')) ||
+        (linkedField.notAfter && linkedDate.isBefore(date, 'day'))
+      ) {
+        updateDate(linkedDate);
+      }
+    }
+    linkedFieldDateElem.addEventListener('change', handleDateChange);
+
+    return () => {
+      linkedFieldDateElem.removeEventListener('change', handleDateChange);
+    };
+  }, [date, linkedFieldDateElem, linkedField, updateDate]);
+
+  const clearFields = () => {
+    updateTime(null);
+    updateDate(null);
+    clearRef.current.dispatchEvent(new Event('indico:closeAutoTooltip'));
+  };
+
+  const getDisabledHours = useCallback(() => {
+    const hours = [];
+    if (earliestMoment && date.isSame(earliestMoment, 'day')) {
+      hours.push([...Array(earliestMoment.hour()).keys()]);
+    }
+    if (latestMoment && date.isSame(latestMoment, 'day')) {
+      const hourInt = latestMoment.hour();
+      hours.push([...Array(23 - hourInt).keys()].map(x => x + hourInt + 1));
+    }
+    return hours.flat();
+  }, [earliestMoment, latestMoment, date]);
+
+  const getDisabledMinutes = useCallback(
+    h => {
+      const minutes = [];
+      if (earliestMoment && date.isSame(earliestMoment, 'day') && h === earliestMoment.hour()) {
+        minutes.push([...Array(earliestMoment.minutes()).keys()]);
+      }
+      if (latestMoment && date.isSame(latestMoment, 'day') && h === latestMoment.hour()) {
+        const minInt = latestMoment.minutes();
+        minutes.push([...Array(59 - minInt).keys()].map(x => x + minInt + 1));
+      }
+      return minutes.flat();
+    },
+    [earliestMoment, latestMoment, date]
+  );
+
+  const isOutsideRange = useCallback(
+    value => {
+      if (linkedField) {
+        const linkedDate = moment(linkedFieldDateElem.value, 'DD/MM/YYYY');
+        if (linkedField.notBefore && value.isBefore(linkedDate, 'day')) {
+          return true;
+        }
+        if (linkedField.notAfter && value.isAfter(linkedDate, 'day')) {
+          return true;
+        }
+      }
+      if (earliestMoment && value.isBefore(earliestMoment, 'day')) {
+        return true;
+      }
+      if (latestMoment && value.isAfter(latestMoment, 'day')) {
```

Comment:
not sure if this improves readability though

Reply:
IMO, we could perhaps simplify this by moving the null checks up and implement the logic with isBefore or After, return true. Since `x && value.isBefore(x, 'day')` is being repeated.

Reply:
Not sure how exactly you mean this, I changed it a bit in the newest version. If you think it's better in a different way please tell me :)

Reply:
Much better.